src/new-client/src/components/DraggableTearOut/DraggableTearOut.tsx

```diff
@@ -0,0 +1,120 @@
+import styled from "styled-components"
+import { RefObject, useState } from "react"
+import { tearOutSection, Section } from "@/App/TearOutSection/state"
+import { canDrag } from "@/components/DraggableTearOut/canDrag"
+import { tearOut } from "@/App/LiveRates/Tile/TearOut/state"
+
+export const DragWrapper = styled.div`
+  height: 100%;
+`
+
+type DraggableSectionTearOutProps = React.PropsWithChildren<{
+  section: Section
+  disabled?: boolean
+}>
+
+type DraggableTileTearOutProps = React.PropsWithChildren<{
+  symbol: string
+  tileRef: RefObject<HTMLDivElement>
+  disabled?: boolean
+}>
+
+type DraggableTearOutGenericProps = React.PropsWithChildren<{
+  dragHandler: () => void
+}>
+
+export const DraggableSectionTearOut: React.FC<DraggableSectionTearOutProps> = (
+  props: DraggableSectionTearOutProps,
+) => {
+  const draggable = canDrag && !props.disabled
+  const dragHandler = () => {
+    tearOutSection(true, props.section)
+  }
+  return (
+    <>
```

Comment:
Another really minor style point - If the top level of a render function is javascript then you don't need to provide the empty `<></>` tags

Reply:
I'm not sure if I follow, whenever I try to remove the encapsulation of the props.children, a typescript error appears:

```
Type '(props: PropsWithChildren<DraggableSectionTearOutProps>) => ReactNode' is not assignable to type 'FC<DraggableSectionTearOutProps>'.
  Type 'ReactNode' is not assignable to type 'ReactElement<any, any> | null'.
    Type 'undefined' is not assignable to type 'ReactElement<any, any> | null'.ts(2322)
```

for now I moved the encapsulation of empty tags be with the props.children

```
return (
    draggable ? (
        <DraggableTearOut dragHandler={dragHandler}>
          {props.children}
        </DraggableTearOut>
      ) : <>{props.children}</>
  )
```